### PR TITLE
[WIP] Run androidTests in Travis

### LINKED ISSUE #2226: Running androidTests in Travis
**Summary:** 

I was thinking about how to write a load of espresso tests (see #586) for the app to try to prevent so many bugs cropping up.

To get good coverage, we'd need to be a logged in user. So the tests would be more standardised, I think it would be useful to have a test user the tests could login as.

Therefore I think it would be useful to have an account on beta commons (and maybe even on real/prod commons) which isn't able to do anything - not even edit its own user talk page, or change its own password. Maybe:
- Create the account
- Upload a few test photographs (so the 'home' page is not blank)
- Get it 'locked' somehow

It also might be useful to have a similar test user which is a completely blank account (i.e. no uploads), as I could imagine that causing bugs in the app which we want to test for (#2154).

We could then safely commit its credentials to this repo so people can run the tests. There also might be a different way that doesn't involve sharing credentials, if people have one please do suggest!

The alternate idea I had:
- Require developers to create their own test accounts according to some specific instructions and then
    - Log in to the app in the emulator and then run the test suite
    - Put their credentials in a file that is not commited to the repo (with some `.gitignore` rule)
- Somehow Travis has its own credentials (kept secret like Google Play keys) and does its testing like that?

The issue with this is I can imagine a lot of new contributors accidentally pushing their credentials to Github. While this would be bad for beta commons, this could be very bad on real/prod commons. The setup would add an extra barrier to entry for developers to start regularly using automated testing, which we want to encourage by making as easy as possible. I also have no idea how to do the bit about Travis.

**Would you like to work on the issue?**

I don't really know how this would be done - hoping someone else here might. If not, I'm happy to post on the Commons helpdesk and see if we can get some answers.

Yes in terms of trying to add more android tests - although they're so broad and don't really conflict everyone is still free to (and should!) get writing tests. I'm not very good at it yet, but I'm trying to learn :)

### PR DESCRIPTION
**Pull request by domdomegg**
**Description**

Fixes #2226, #2823. Has Travis run all the android tests.

An example is here:
https://travis-ci.com/domdomegg/apps-android-commons/builds/108990409

~~@maskaravivek Due to #2868 this doesn't work on master - probably need your help with this. Having a look at #2829 the obvious cause is the removal of StringUtils but using alternatives just doesn't seem to work after that PR (e.g. even just checking if it's an empty string).~~

**Future plans**

Ideally the androidTests can be separated from the unit tests for faster builds. I think this should be relatively easy once this is up and running on Travis. Relevant docs: https://docs.travis-ci.com/user/build-matrix/

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -110,7 +110,6 @@ android {
         minSdkVersion 19
         targetSdkVersion 28
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
-        testInstrumentationRunnerArguments clearPackageData: 'true'
```
Comment (maskaravivek) [inline app/build.gradle]: Will all tests pass if we don't clear data before every test?

Comment (domdomegg) [reply app/build.gradle]: Yes, they all work fine now as we've removed the temperamental ones, and the emulator starts it from a clean state at the beginning.

**Comment by domdomegg** [comment]
@maskaravivek please can you have a look at fixing #2868 so this can be merged.

**Comment by maskaravivek** [comment]
@domdomegg I have fixed #2868. Let us try to get this PR merged. :)

**Comment by maskaravivek** [comment]
@domdomegg Is the PR ready for review?

**Comment by domdomegg** [comment]
Don't think so, Travis still isn't building it :(

**Comment by maskaravivek** [comment]
Yup, i had tried using your changes in #3298 but faced the same issue.

**Comment by neslihanturan** [comment]
This PR is WIP for long time, do you think we should keep it open @domdomegg , or do you think another solution will be needed for this task?

**Comment by domdomegg** [comment]
We need to redo some of the UI test logic I think. Will require more changes. Closing for now.

**Pull request closed without merging**